src/fsspec_xrootd/xrootd.py

```diff
@@ -43,39 +61,184 @@ def _get_kwargs_from_urls(u: str) -> dict[Any, Any]:
+    def rmdir(self, path: str) -> None:
+        status, n = self._myclient.rmdir(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"Directory not removed properly: {status.message}")
 
-        listing = []
+    def _rm(self, path: str) -> None:
+        status, n = self._myclient.rm(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"File not removed properly: {status.message}")
 
-        if detail:
-            for item in deets:
-                t = ""
-                if item.statinfo.flags and StatInfoFlags.IS_DIR:
-                    t = "directory"
-                elif item.statinfo.flags and StatInfoFlags.OTHER:
-                    t = "other"
+    def touch(self, path: str, truncate: bool = False, **kwargs: Any) -> None:
+        if truncate or not self.exists(path):
+            status, _ = self._myclient.truncate(path, 0, timeout=self.timeout)
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+        else:
+            status, _ = self._myclient.truncate(
+                path, self.info(path).get("size"), timeout=self.timeout
+            )
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+
+    def modified(self, path: str) -> Any:
+        status, statInfo = self._myclient.stat(path, timeout=self.timeout)
+        return statInfo.modtime
+
+    def sign(self, path: str, expiration: int = 100, **kwargs: Any) -> Any:
+        return (
+            self.storage_options["protocol"]
+            + "://"
+            + self.storage_options["hostid"]
+            + "/"
+            + self.storage_options["path_with_params"]
+        )
+
+    def exists(self, path: str, **kwargs: Any) -> bool | Any:
+        if path in self.dircache:
+            return True
+        else:
+            status, _ = self._myclient.stat(path, timeout=self.timeout)
+            if not status.ok:
+                if status.code == 400:
+                    return False
                 else:
-                    t = "file"
-
-                listing.append(
-                    {
-                        "name": path + "/" + item.name,
-                        "size": item.statinfo.size,
-                        "type": t,
-                    }
-                )
+                    raise OSError(f"status check failed with message: {status.message}")
+            else:
+                return True
+
+    def info(self, path: str, **kwargs: Any) -> dict[str, Any]:
+        spath = os.path.split(path)
+        deet = self._ls_from_cache(spath[0])
+        if deet is not None:
+            det = []
+            for item in deet:
+                if item["name"] == path:
+                    det.append(item)
+            return {
+                "name": path,
+                "size": det[0]["size"],
+                "type": det[0]["type"],
+            }
```

Comment:
Here you could just return the first item that matches rather than loop through all entries
```
for item in deet:
    if item["name"] == path:
        return item
```

Reply:
I agree with your change. After making the change, pre-commit demanded that I add a return of some sort outside the for loop in case the if statement is never satisfied. I'm having it raise an error in that case, although I see no reason this error should ever be raised.